sphericart/CMakeLists.txt

```diff
@@ -1,6 +1,43 @@
 cmake_minimum_required(VERSION 3.10)
 
 project(sphericart LANGUAGES C CXX)
+#R"======(, )======"
```

Comment:
templates_core.hpp was separated from templates.hpp, since 1) we only need these specific functions in the CUDA code and 2) compilation with nvrtc will fail horribly if we try to include the original templates.hpp into the cuda_base.hpp.

cuda_base.hpp is the original front-end to the CUDA code, but now it calls nvrtc instead.

sphericart_cuda.hpp is the code which deals with the sphericart::cuda classes.